### ISSUE TITLE
[err:FODC0002] "http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml" (Line 1): Premature end of file. 

### ISSUE THREAD

**Issue by christian-schigl** [INSPIRE-MIF/helpdesk-validator#23]
Before reporting an issue with the INSPIRE validator deployed at JRC, please check the [list of known limitations](https://github.com/inspire-eu-validation/community/wiki/Known-limitations).

_=> Just saw that this issue already has been adressed. Nevertheless, I couldn't find how to solve it.
Is there a workaround ?

cheers christian_
 
Please replace these lines with a brief summary of your issue and the expected
behaviour.

To help us investigate the issue, you may want to include:

- [ ] a description of the steps to reproduce your issue
ETF-Validator : Start Tests / Metadata / Conformance class: INSPIRE Profile based on EN ISO 19115 and EN ISO 19119 + Metadata for interoperability
- [ ] the browser and version used
Firefox Quantum 60.4
- [ ] a screenshot illustrating the problem
![grafik](https://user-images.githubusercontent.com/47251949/52134230-62935d80-2643-11e9-9bfd-3d35287e347f.png)

- [ ] the Test Report (ideally as an attachment to the issue)
.. see attached file
- [ ] the name of the Test that failed or better: change the `Level of Detail`
in the Report by clicking on `All Details`, then scroll down to the failed test
and copy the Link on the right-hand side of the `Assertion URI`
http://inspire-sandbox.jrc.ec.europa.eu/etf-webapp//v2/TestRuns/EID19318236-77e9-4dc3-bf89-73c071a6666f.html?lang=en#EID816e17b5-0818-4225-b18e-67c3d36bfcda

http://inspire-sandbox.jrc.ec.europa.eu/etf-webapp//v2/TestRuns/EID19318236-77e9-4dc3-bf89-73c071a6666f.html?lang=en#EIDd5808fef-76f3-4646-8dca-b0403cc09e96

- [ ] the URL of your service / the file you have uploaded or referenced (as
an attachment).
.. see attached file
[ETF_Error.zip](https://github.com/inspire-eu-validation/community/files/2822374/ETF_Error.zip)

If your service is protected or you have confidential data that you cannot
upload here, you can request an email address to which you can send URLs / files.
Please note that issues cannot be investigated for services/datasets that we cannot
access.

**Comment by inakidiazdecerio** [comment]
Dear @christian-schigl, 

Thanks for reporting the issue. The issue has been already registered and the solution has been merged into master branch, but is pending to be deployed in reference validator. 

Regards,
Iñaki

**Comment by MarcoMinghini** [comment]
This is the issue which has been already addressed and solved: https://github.com/inspire-eu-validation/community/issues/5

**Comment by inakidiazdecerio** [comment]
@christian-schigl we have made some improvements in the code to call to the https secure urls instead of the http to avoid the redirection and solve the "Premature end of file" error. There's no need to change anything in your metadata.

**Comment by inakidiazdecerio** [comment]
@christian-schigl the issue is pending to be deployed in the production environment. I'll let you know when the issue is fixed.

**Comment by MarcoMinghini** [comment]
Dear @christian-schigl, a solution to the issue has been deployed for testing in the new Staging instance of the INSPIRE Reference Validator, which is available at http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/

I tested the metadata that generated the issue and I did not get any error. However, can you also try to validate again the metadata and provide a feedback?
Thank you.

**Comment by christian-schigl** [comment]
Dear Marco,

i just ran a test of the attached Metadata.

It ends up in 2 failed categories, namely
- character encoding
- spatial representation types
both stating that
„System error in the Executable Test Suite. Please contact a system administrator. Error information:
[err:FODC0002] "http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml" (Line 1): Premature end of file.
 (495/29)“

As You can see in the attached file this URL is never refenced – only https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/Codelist/ML_gmxCodelists.xml

So from my point of view the problem is not yet solved,

sincerly
Christian



Von: Marco Minghini *EXTERN* [mailto:notifications@github.com]
Gesendet: Mittwoch, 17. April 2019 10:11
An: inspire-eu-validation/community
Cc: Schigl Christian; Mention
Betreff: Re: [inspire-eu-validation/community] [err:FODC0002] "http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml" (Line 1): Premature end of file. (#23)


Dear @christian-schigl<https://github.com/christian-schigl>, a solution to the issue has been deployed for testing in the new Staging instance of the INSPIRE Reference Validator, which is available at http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/

I tested the metadata that generated the issue and I did not get any error. However, can you also try to validate again the metadata and provide a feedback?
Thank you.

**Comment by inakidiazdecerio** [comment]
Dear @christian-schigl,

We tested the metadata that generated this issue in the Staging environment and we didn't get any error in character encoding or spatial representation type (see the attached image).

Have you tested in the Staging instance or in Production?

The Staging environment is available at: http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/

Regards.

![image](https://user-images.githubusercontent.com/36446028/56658736-a5416080-669b-11e9-86dd-00afacf7e57e.PNG)

**Comment by christian-schigl** [comment]
No, i have tested in the production environement …

Tried it again in the staging instance – no more errors !

Thank you,
Christian Schigl

Von: notifications@github.com *EXTERN* [mailto:notifications@github.com]
Gesendet: Mittwoch, 24. April 2019 14:20
An: inspire-eu-validation/community
Cc: Schigl Christian; Mention
Betreff: Re: [inspire-eu-validation/community] [err:FODC0002] "http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/ML_gmxCodelists.xml" (Line 1): Premature end of file. (#23)


Dear @christian-schigl<https://github.com/christian-schigl>,

We tested the metadata that generated this issue in the Staging environment and we didn't get any error in character encoding or spatial representation type (see the attached image).

Have you tested in the Staging instance or in Production?

The Staging environment is available at: http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/

Regards.

[Das Bild wurde vom Absender entfernt. image]<https://user-images.githubusercontent.com/36446028/56658736-a5416080-669b-11e9-86dd-00afacf7e57e.PNG>
